raiden/network/transport/matrix/transport.py

```diff
@@ -557,27 +559,42 @@ def whitelist(self, address: Address) -> None:
         # the node that has to create the room.
         self._maybe_create_room_for_address(address)
 
-    def start_health_check(self, node_address: Address) -> None:
+    def async_start_health_check(self, node_address: Address) -> None:
         """Start healthcheck (status monitoring) for a peer
 
         It also whitelists the address to answer invites and listen for messages
         """
-        self.whitelist(node_address)
-        with self._health_lock:
-            node_address_hex = to_normalized_address(node_address)
-            self.log.debug("Healthcheck", peer_address=to_checksum_address(node_address))
+        self._healthcheck_queue.put(node_address)
 
-            candidates = self._client.search_user_directory(node_address_hex)
-            self._displayname_cache.warm_users(candidates)
+    def immediate_health_check_for(self, node_address: Address) -> None:
+        """Start healthcheck (status monitoring) for a peer
+
+        It also whitelists the address to answer invites and listen for messages
+        """
+        self.whitelist(node_address)
```

Comment:
Humm, but that was the point of the PR. What races do you expect to happen?

Reply:
Yeah as discussed in person it probably shouldn't hurt.

Just to document it:
With the whitelisting being performed asynchronously we could theoretically receive a message from the to-be-whitelisted node before the whitelisting has happened. But since we have retry this should not cause a bug.

Reply:
> With the whitelisting being performed asynchronously we could theoretically receive a message from the to-be-whitelisted node before the whitelisting has happened. But since we have retry this should not cause a bug.

Additional note, this would be particularly bad for one hop transfers.